R/build_html_mans.R

```diff
@@ -0,0 +1,187 @@
+#'
+#'     saveRDS(meta_aliases_db, aliases_db_file, version = 2)
+#' }
+#' @export
+build_meta_aliases_db <-
+    function(web_dir, aliases_db_file, force = FALSE)
+{
+    files <- Sys.glob(file.path(web_dir, "*", "aliases.rds"))
+    packages <- basename(dirname(files))
+    if (force || !is_file(aliases_db_file)) {
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(aliases_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which aliases file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(aliases_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+is_file <- function(x) file.exists(x) && !file.info(x)[["isdir"]]
+
+#' @rdname html-manuals
+#'
+#' @param rdxrefs_db_file `character(1)` The file path to `rdxrefs.rds` file
+#'   generated by the `build_db_from_source` function.
+#'
+#' @examples
+#' if (interactive()) {
+#'     src_base <- "~/minibioc/packages/3.20/bioc/"
+#'     web_dir <- file.path(src_base, "web", "packages")
+#'
+#'     meta_folder <- file.path(contrib.url(src_base), "Meta")
+#'     if (!dir.exists(meta_folder)) dir.create(meta_folder, recursive = TRUE)
+#'     rdxrefs_db_file <- file.path(meta_folder, "rdxrefs.rds")
+#'
+#'     meta_rdxrefs_db <- build_meta_rdxrefs_db(web_dir, rdxrefs_db_file)
+#'
+#'     saveRDS(meta_rdxrefs_db, rdxrefs_db_file, version = 2)
+#' }
+#' @export
+build_meta_rdxrefs_db <-
+    function(web_dir, rdxrefs_db_file, force = FALSE)
+{
+    files <- Sys.glob(file.path(web_dir, "*", "rdxrefs.rds"))
+    packages <- basename(dirname(files))
+    if(force || !is_file(rdxrefs_db_file)) {
+        db <- lapply(files, readRDS)
+        names(db) <- packages
+    } else {
+        db <- readRDS(rdxrefs_db_file)
+        ## Drop entries in db not in package web area.
+        db <- db[!is.na(match(names(db), packages))]
+        ## Update entries for which rdxrefs file is more recent than the
+        ## db file.
+        mtimes <- file.mtime(files)
+        files <- files[mtimes >= file.mtime(rdxrefs_db_file)]
+        db[basename(dirname(files))] <- lapply(files, readRDS)
+    }
+
+    db[sort(names(db))]
+}
+
+#' @rdname html-manuals
+#'
+#' @param package_dirs `character()` A vector of local package directories.
+#'
+#' @examples
+#' if (interative()) {
+#'     library(BiocPkgTools)
+#'     bioc_sub <- pkgBiocDeps(
+#'         "SummarizedExperiment", pkgType = "software",
+#'         recursive = TRUE, only.bioc = TRUE
+#'     )
+#'     bioc_sub <- unlist(bioc_sub, use.names = FALSE)
+#'
+#'     ## generate from Bioc package source dirs
+#'     packages <- file.path(normalizePath("~/bioc"), bioc_sub)
+#'     src_base <- "~/minibioc/packages/3.20/bioc"
```

Comment:
I think line 170 is a little confusing in the context of the CRAN-style repo as manuals don't reside there. (I know this is just an example, but I thought was part of the "instructions" before.) It might be better to call it `destDir`, which is what is called in another location: https://github.com/Bioconductor/biocViews/blob/8aa6dafe1f6f65a55ff5dd6f3565690c49a5618e/R/repository.R#L61.

Reply:
> I think line 170 is a little confusing in the context of the CRAN-style repo as manuals don't reside there.

But the manuals are in 
https://bioconductor.org/packages/release/bioc/manuals/MultiAssayExperiment/man/MultiAssayExperiment.pdf
I'm not sure I understand what you mean. 

`src_base` is more descriptive than `destDir` since I differentiate between source and binary packages in the examples. Does `src_destDir` work ?

Reply:
You're right; my apologies. I'm getting some BBS imposed structures mixed up with the CRAN structures.

Thinking about your original code, I thought it should be used to construct the HTML in a temporary place that had `manuals` in the path whereas now, you create and place in what should be their final structure. Now `src_base` seems more fitting but `src_destDir` is okay.